app/grandchallenge/reader_studies/serializers.py

```diff
@@ -95,15 +96,38 @@ class AnswerSerializer(HyperlinkedModelSerializer):
     )
 
     def validate(self, attrs):
-        question = attrs["question"]
-        images = attrs["images"]
-        answer = attrs["answer"]
-        creator = self.context.get("request").user
+        answer = attrs.get("answer")
+        if self.instance:
+            if (
+                not self.instance.question.reader_study.allow_answer_modification
+            ):
+                return {}
+            question = self.instance.question
+            images = self.instance.images.all()
+            creator = self.instance.creator
+        else:
+            question = attrs.get("question")
+            images = attrs.get("images")
+            creator = self.context.get("request").user
 
         Answer.validate(
-            creator=creator, question=question, answer=answer, images=images
+            creator=creator,
+            question=question,
+            answer=answer,
+            images=images,
+            instance=self.instance,
         )
-        return attrs
+
+        if self.instance:
+            add_scores.apply_async(
+                kwargs={
+                    "instance_pk": str(self.instance.pk),
+                    "pk_set": list(
+                        map(str, images.values_list("pk", flat=True))
+                    ),
+                }
+            )
```

Comment:
This is what I thought at first as well, but the problem is that we cannot do that, because m2m relations get added after the save. So if we call this post save, there are no images set for the answer and we cannot calculate the score. We could do this for the update of course, but I felt this was a bit nicer than adding another signal handler. I'd be happy to change it if you disagree though.

Reply:
Thanks, wasn't clear to me. I agree, we should avoid signal handlers at all costs but it's impossible to do that for m2m relationships.